factory/factory.go

```diff
@@ -12,128 +12,48 @@ package factory
 
 import (
 	"fmt"
+	"net/url"
 	"os"
-	"reflect"
+	"sync"
 
 	"github.com/omec-project/amf/logger"
 	"gopkg.in/yaml.v2"
 )
 
-var AmfConfig Config
+var (
+	AmfConfig  Config
+	ConfigLock sync.RWMutex
```

Comment:
Given that some of the parameters were moved to the `context`, ss this variable still needed?